app/src/ui/dispatcher/dispatcher.ts

```diff
@@ -1687,6 +1687,12 @@ export class Dispatcher {
     // up-to-date before performing the "Clone in Desktop" steps
     await this.appStore._refreshRepository(repository)
 
+    // if the repo has a remote, fetch before switching branches to ensure
+    // the checkout will be successful
+    if (isRepositoryWithGitHubRepository(repository)) {
+      await this.appStore._fetch(repository, FetchType.UserInitiatedTask)
+    }
```

Comment:
So this is rather nitpicky but `isRepositoryWithGitHubRepository` checks whether the Repository is hosted on GitHub.com/GHES/GHAE vs some other Git server. Is there a compelling reason to limit this to just GitHub repositories?

```suggestion
    await this.appStore._fetch(repository, FetchType.UserInitiatedTask)
```

Reply:
@niik I don't think that's a nitpick at all 😄 ! We dont' have any reason to limit this to just GitHub repositories.  I think I modeled the type signature of [fetch_default_remote](https://github.com/desktop/desktop/pull/13641/commits/ffc6d887a9b9259b78f649a0361dbce5f6039475#diff-b238c18a99c08ab800e4227) (from my original commit) off of a function that dealt with a pull request and conflated "fetching" with "remotes". 

I'm glad you pointed it out, and thanks for reading carefully.   I've just added a commit to eliminate the 'if' check.

Reply:
Awesome, and thanks for tweaking that comment there as well, I love it!